internal/venice-common/src/main/java/com/linkedin/venice/utils/Utils.java

```diff
@@ -632,6 +644,29 @@ private static String getRealTimeTopicNameIfEmpty(String realTimeTopicName, Stri
     return Strings.isBlank(realTimeTopicName) ? composeRealTimeTopic(storeName) : realTimeTopicName;
   }
 
+  public static String createNewRealTimeTopicName(String oldRealTimeTopicName) {
```

Comment:
Can we define a standard format for topic names. This seems frail  approach to me. If we want to use versioning for RT topics then shouldn't we store those in store config? Also how would we know the partition count of RTs without querying the PubSub?

Reply:
This is quite deterministic format, do you have a better way?
I am not really trying to support versioning of RT in the sense that at one time only one rt should exist, unless a replicator is running, in which case it should be maximum 2 topics at a time.
Yes, querying the pubsub topic is the way to know that, is it not a good way? We would not require to make a lot of such queries.

Reply:
> This is quite deterministic format, do you have a better way?

If we take `storeName_v1` as a store name and `storeName_v11_v55_rt` as the old topic, the new expected RT topic name for this store should be `storeName_v11_v56_rt`, correct? However, if I understand correctly, this method would return `storeName_v11_v55_v2_rt`. Is that the intended behavior? It doesn’t seem right to me.


Maybe you want something like this:
```java
public static String createNewRealTimeTopicName(String oldRealTimeTopicName) {
    // Validate input
    if (oldRealTimeTopicName == null || !oldRealTimeTopicName.endsWith(Version.REAL_TIME_TOPIC_SUFFIX)) {
        throw new IllegalArgumentException("Invalid old name format");
    }

    // Find the base name by removing the suffix
    int suffixLength = Version.REAL_TIME_TOPIC_SUFFIX.length();
    String base = oldRealTimeTopicName.substring(0, oldRealTimeTopicName.length() - suffixLength);

    // Locate the last version separator "_v" in the base
    int versionSeparatorIndex = base.lastIndexOf("_v");
    if (versionSeparatorIndex > -1 && versionSeparatorIndex < base.length() - 2) {
        // Extract and increment the version
        String versionStr = base.substring(versionSeparatorIndex + 2);
        int version = Integer.parseInt(versionStr) + 1;
        base = base.substring(0, versionSeparatorIndex) + "_v" + version;
    } else {
        // Start with version 2 if no valid version is present
        base = base + "_v2";
    }

    return base + Version.REAL_TIME_TOPIC_SUFFIX;
}
```

Reply:
> Yes, querying the pubsub topic is the way to know that, is it not a good way? We would not require to make a lot of such queries.


I should we should have our own mapping of partition count to rt name somewhere. It could be either via convention or some mapping somewhere in the store config